Sources/StreamChat/Workers/Background/ConnectionRecoveryUpdater.swift

```diff
@@ -95,87 +95,61 @@ class ConnectionRecoveryUpdater: EventWorker {
             
-            if useSyncEndpoint {
-                self?.sync(completion: refetchExistingQueries)
+            if self?.useSyncEndpoint == true {
+                let cids = Set(
+                    context
+                        .loadChannelListQueries()
+                        .flatMap(\.channels)
+                        .map(\.channelId)
+                        .prefix(1000)
+                )
+                
+                self?.getMissingEvents(for: cids, since: lastSyncedAt) { error in
+                    self?.databaseCleanupUpdater.syncChannelListQueries(
+                        syncedChannelIDs: error == nil ? cids : []
+                    )
+                }
             } else {
-                refetchExistingQueries()
+                self?.databaseCleanupUpdater.syncChannelListQueries(syncedChannelIDs: [])
             }
         }
     }
-
-    private func sync(completion: @escaping () -> Void) {
-        guard let lastSyncedAt = lastSyncedAt else { return }
-        
-        let watchedChannelIDs = allChannels.map(\.cid).compactMap { try? ChannelId(cid: $0) }
-        
-        guard !watchedChannelIDs.isEmpty else {
-            log.info("Skipping `/sync` endpoint call as there are no channels to watch.")
-            return
-        }
-        
-        let endpoint: Endpoint<MissingEventsPayload> = .missingEvents(
-            since: lastSyncedAt,
-            cids: watchedChannelIDs
-        )
+    
+    private func getMissingEvents(
+        for cids: Set<ChannelId>,
+        since lastSyncedAt: Date,
+        completion: @escaping (Error?) -> Void
+    ) {
+        log.debug("Will fetch missing events for \(cids) starting from \(lastSyncedAt)")
         
-        apiClient.request(endpoint: endpoint) { [weak self] in
-            guard let self = self else { return }
+        apiClient.request(
+            endpoint: .missingEvents(since: lastSyncedAt, cids: .init(cids))
+        ) { [weak self] in
             switch $0 {
             case let .success(payload):
-                // The sync call was successful.
-                // We schedule all events for existing channels for processing...
-                self.eventNotificationCenter.process(payload.eventPayloads)
-                
-                // ... and refetch the existing queries to see if there are some new channels
-                completion()
+                log.debug("Did receive \(payload.eventPayloads.count) missing events: \(payload.eventPayloads)")
                 
+                self?.eventNotificationCenter.process(payload.eventPayloads) {
+                    completion(nil)
+                }
             case let .failure(error):
-                log.info(
-                    """
-                    Backend couldn't handle replaying missing events - there was too many (>1000) events to replay. \
-                    Cleaning local channels data and refetching it from scratch
-                    """
```

Comment:
So we don't have this mechanism anymore? on error, we just keep the channel data around?